v6_carrier_py/master.py

```diff
@@ -105,12 +112,53 @@ def correlation_matrix(client: ClientContainerProtocol, data, keys=None, *args,
     If no keys are specified the datasets are joined on all columns with the same name.
     TODO: What if different datasets use different keys to mean the same thing? How do we specify this?
     """
-    results = _dispatch_tasks(client, data, *args, method='get_data', **kwargs)
+    combined_df = _combine_all_node_data(client, data, keys, *args, **kwargs)
+
+    return combined_df.corr()
+
+
+def train_model(client: ClientContainerProtocol, data, pipeline: Pipeline, features: List[str], target: str, keys=None,
+                *args, **kwargs):
+    """
+    Retrieve data from nodes and train data analysis pipeline on it. Returns the performance of the resulting model.
+    TODO: How and where do we save our model?
```

Comment:
I guess you want to return the model? But I guess it is hard to serialize. Maybe store it somewhere and return a download link (might also need authentication)

Reply:
I don't know if I want to return the model actually, It seems like something that is hard to understand whether it infringes on someones privacy. I'd rather keep it at the TSE and just return the performance, plus maybe an ID of the model you just ran. Then if the performance is satisfactory you can run another algorithm to use this model on other data.

I'm not 100% convinced about this though, I think I need to read up some more about federated learning.

Reply:
As an example of a mode infringing on someones privacy. There could be an overfitted decision tree where one of the leafs only fits one person in the training set. Looking a the model could tell you a lot about that individual.

Reply:
We could also return an encrypted version of the model, where the user can give it as input to an algorithm but can't decrypt it themselves.